tests/konflux-demo/konflux-demo.go

```diff
@@ -40,464 +39,378 @@ import (
+			It("creates an IntegrationTestScenario for the app", Label(devEnvTestLabel), func() {
+				its := appSpec.ComponentSpec.IntegrationTestScenario
+				integrationTestScenario, err = fw.AsKubeAdmin.IntegrationController.CreateIntegrationTestScenario("", appSpec.ApplicationName, fw.UserNamespace, its.GitURL, its.GitRevision, its.TestPath)
+				Expect(err).ShouldNot(HaveOccurred())
+			})
 
-				// Component are imported from gitUrl
-				It(fmt.Sprintf("creates component %s (private: %t) from git source %s", componentSpec.Name, componentSpec.Private, componentSpec.GitSourceUrl), Label(devEnvTestLabel, stageEnvTestLabel), func() {
-
-					componentObj := appservice.ComponentSpec{
-						ComponentName: componentSpec.Name,
-						Application:   appTest.ApplicationName,
-						Source: appservice.ComponentSource{
-							ComponentSourceUnion: appservice.ComponentSourceUnion{
-								GitSource: &appservice.GitSource{
-									URL:           componentSpec.GitSourceUrl,
-									Revision:      gitRevision,
-									Context:       componentSpec.GitSourceContext,
-									DockerfileURL: componentSpec.DockerFilePath,
-								},
+			It("creates new branch for the build", Label(devEnvTestLabel), func() {
+				gitRevision = appSpec.ComponentSpec.GitSourceRevision
+				// We need to create a new branch that we will target
+				// and that will contain the PaC configuration, so we
+				// can avoid polluting the default (main) branch
+				componentNewBaseBranch = fmt.Sprintf("base-%s", util.GenerateRandomString(6))
+				gitRevision = componentNewBaseBranch
+				Expect(fw.AsKubeAdmin.CommonController.Github.CreateRef(componentRepositoryName, appSpec.ComponentSpec.GitSourceDefaultBranchName, appSpec.ComponentSpec.GitSourceRevision, componentNewBaseBranch)).To(Succeed())
+			})
+
+			// Component are imported from gitUrl
+			It(fmt.Sprintf("creates component %s (private: %t) from git source %s", appSpec.ComponentSpec.Name, appSpec.ComponentSpec.Private, appSpec.ComponentSpec.GitSourceUrl), Label(devEnvTestLabel), func() {
+				componentObj := appservice.ComponentSpec{
+					ComponentName: componentName,
+					Application:   appSpec.ApplicationName,
+					Source: appservice.ComponentSource{
+						ComponentSourceUnion: appservice.ComponentSourceUnion{
+							GitSource: &appservice.GitSource{
+								URL:           appSpec.ComponentSpec.GitSourceUrl,
+								Revision:      gitRevision,
+								Context:       appSpec.ComponentSpec.GitSourceContext,
+								DockerfileURL: appSpec.ComponentSpec.DockerFilePath,
 							},
 						},
+					},
+				}
+
+				component, err = fw.AsKubeAdmin.HasController.CreateComponent(componentObj, namespace, "", secret.Name, appSpec.ApplicationName, false, utils.MergeMaps(constants.ComponentPaCRequestAnnotation, constants.DefaultDockerBuildPipelineBundle))
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+
+			When("Component is created", func() {
+
+				It("triggers creation of a PR in the sample repo", func() {
+
```

Comment:
```suggestion
				It("triggers creation of a PR in the sample repo", func() {
```
can we remove these new lines for consistency?